src/workloads/query/PercentilesSpillToDisk.yml

```diff
@@ -0,0 +1,145 @@
+  when it spills to disk. There are a few cases to consider
+  1. Nothing spills.
+  2. Only the $_internalSetWindowsFields will spill, which in this case would be the $percentile
+  window function.
+  3. Both the $sort and $_internalSetWindowsFields spill.
+  There is a workload in SetWindowFieldsUnbounded.yml that handles the case if only the $sort spills,
+  and since there is no unique behavior for $percentile in this case, it was not tested in this file.
+  We can use an index to control whether or not the $sort will spill, and we change the size of the
+  partition to control whether or not the $_internalSetWindowsFields will spill.
+
+Keywords:
+- setWindowFields
+- percentile
+- spill
+
+AutoRun:
+- When:
+    mongodb_setup:
+      $eq:
+      - standalone-all-feature-flags
+    branch_name:
+      $neq:
+      - v4.0
+      - v4.2
+      - v4.4
+      - v5.0
+      - v6.0
+
+GlobalDefaults:
+  MaxPhases: &maxPhases 8
+  Database: &db percentiles
+  Collection: &coll Collection0
+
+ActorTemplates:
+- TemplateName: ComputePercentile
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "ComputePercentile"}}
+    Type: CrudActor
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          Repeat: 10
+          Database: *db
+          Collection: *coll
+          Operations:
+          - OperationName: aggregate
+            OperationCommand:
+              Pipeline: [{
+                $setWindowFields: {
+                  partitionBy: {^Parameter: {Name: "Partition", Default: null}},
+                  sortBy: {^Parameter: {Name: "Sort", Default: {not_spill_indexed: 1}}},
+                  output: {
+                    p: {
+                      $percentile: {p: [0.75], input: "$data", method: "approximate" },
+                      window: { documents: [0, "unbounded"] }
+                    }}
+                }}
+              ]
+              Options:
+                AllowDiskUse: {^Parameter: {Name: "Disk", Default: true}}
+
+Actors:
+- Name: CreateDataset
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *maxPhases
+      PhaseConfig:
+        Threads: 1
+        Repeat: 1
+        Database: *db
+        CollectionCount: 1
+        # We have 100k documents with each being between 10kB to 20kB. When divided into 20 partitions,
+        # it won't spill, but when divided into fewer partitions (like 2) it will spill.
+        DocumentCount: 100000
+        BatchSize: 1000
+        Document:
+          _id: {^Inc: {start: 0}}
+          # This data field is what we calculate the percentiles of.
+          data: {^RandomDouble: {distribution: normal, mean: 50.0, sigma: 10.0}}
+          # This string pads the doc size out to around 10KB.
```

Comment:
Yes I can add the comment. Based on my understanding spilling can happen in the window function (in this case if we have too many double values) or in the `setWindowFields` infrastructure when we are partitioning the data. However, there is not a difference in the spilling logic. In other words, both test are looking at the same spilling behavior and the same behavior for reading documents from disk. In one scenario we are spilling less but larger documents and the other we are spilling more but smaller documents. Because we are partitioning the data, part of the window function is iterating through the partition and this is what is spilling in this case.

However, we can change the test to have the window function itself spill the data.